core/prisma/migrations/20241017223037_add_new_membership_tables/migration.sql

```diff
@@ -0,0 +1,123 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `role` on the `member_groups` table. All the data in the column will be lost.
+
+*/
+-- AlterTable
+ALTER TABLE "member_groups" DROP COLUMN "role";
+
+-- CreateTable
+CREATE TABLE "community_memberships" (
+    "id" TEXT NOT NULL DEFAULT gen_random_uuid(),
+    "role" "MemberRole" NOT NULL,
+    "communityId" TEXT NOT NULL,
+    "userId" TEXT,
+    "memberGroupId" TEXT,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    CHECK ( num_nonnulls("userId", "memberGroupId") = 1 ),
```

Comment:
These check constraints ensure that we set exactly one of `userId` or `memberGroupId`. We can't express this constraint in the prisma schema, but it should probably be documented somewhere more discoverable than this specific migration.

Reply:
yeah shame prisma just cant do that. I think a comment on the table pointing to this file would be good enough!

`drizzle` seems to have added check constraints recently... (https://orm.drizzle.team/docs/indexes-constraints#check)

May be worth considering using them for our migrations instead at some point, i've grown a bit annoyed at the constraints (hah) of prismas model.

Reply:
added the comment! 

i'd love to replace prisma with something that lets us express all of our constraints in the schema. do you know if drizzle can generate/manage partial indices (an index with a where clause)?